src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -55,9 +115,9 @@ internal sealed class JsonPropertyInfo<T> : JsonPropertyInfo
                 JsonTypeInfo = jsonTypeInfo;
             }
 
-            ConverterBase = converter;
+            NonCustomConverter = converter;
```

Comment:
Maybe something like
```suggestion
            TypeLevelConverter = converter;
```